projects/core/src/routing/url-matcher/default-url-matcher-factory.ts

```diff
@@ -0,0 +1,29 @@
+import { inject, InjectionToken } from '@angular/core';
+import { Route } from '@angular/router';
+import { RoutingConfigService } from '../configurable-routes';
+import { UrlMatcherService } from '../services/url-matcher.service';
+import { UrlMatcherFactory } from './url-matcher-factory';
+
+export function getDefaultUrlMatcherFactory(
+  routingConfigService: RoutingConfigService,
+  urlMatcherService: UrlMatcherService
+): UrlMatcherFactory {
+  const factory = (route: Route) => {
+    const routeName = route.data && route.data['cxRoute'];
+    const routeConfig = routingConfigService.getRouteConfig(routeName);
+    const paths = (routeConfig && routeConfig.paths) || [];
+    return urlMatcherService.fromPaths(paths);
+  };
+  return factory;
+}
+
+export const DEFAULT_URL_MATCHER_FACTORY = new InjectionToken<
```

Comment:
Do we need "FACTORY" in the name here? What about `URL_MATCHER_TOKEN`? If we come up with more specific matchers, we can name them `PRODUCT_URL_MATCHER_TOKEN`.

Reply:
I'm not sure I understand why `TOKEN` is better than `FACTORY`?

I'd like to avoid naming it simply `xxx_URL_MATCHER`, since it can be confusing - it doens't provide an `UrlMatcher`, but a higher order function `Route => UrlMatcher` (`type UrlMatcherFactory`).
I admit the ending is a bit long: `xxx_URL_MATCHER_FACTORY`.
It has a similar length to `xxx_URL_MATCHER_TOKEN` and it describes well that it provides a factory function (not final `UrlMatcher`).

Reply:
I've read your second comment https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/6618#discussion_r384407901 . 
After second thought, I like shorter names, since they are easier to explain to customers:
```
DEFAULT_URL_MATCHER
PRODUCT_LISTING_URL_MATCHER
PRODUCT_DETAILS_URL_MATCHER // using word 'details' for consistency with other places
```